core/src/js/components/main-nav/main-nav.js

```diff
@@ -6,22 +6,23 @@ document.addEventListener('DOMContentLoaded', event => {
 
   // The css class that this following behaviour is applied to.
   const navClass = 'su-main-nav';
-
-  // Variables.
-
   // All main navs.
   const navs = document.querySelectorAll('.' + navClass);
 
-  // ---------------------------------------------------------------------------
-  // Execution Code.
-  // ---------------------------------------------------------------------------
-
   // Process each nav.
   let firstZindex;
   navs.forEach((nav, index) => {
+    // remove the class that formats the nav for browsers with javascript disabled
+    nav.classList.remove('no-js');
+
+    // create an instance of Nav, which in turn creates appropriate instances of NavItem
     const theNav = new Nav(nav);
+
+    // remember the nav for closeAllMobileNavs()
     theNavs.push(theNav);
 
+    // manage zindexes in case there are multiple navs near enough for subnavs to overlap
+    // extremely unlikely, but it happens in the styleguide
```

Comment:
```suggestion
    // extremely unlikely, but it happens in the styleguide.
```

Reply:
If I were totally unfamiliar with Decanter's nav, but reading through the code in anticipation of using it in a project, I would be deeply puzzled by that zIndex loop. I would be reassured by knowing that the problem it's trying to solve is in fact extremely unlikely in production.

Reply:
If you feel strongly about this comment then I am fine to leave it along. My experience in working with open source projects is that I shouldn't try to guess what crazy scenarios another developer may have. Forces unknown can lead to wacky scenarios.

Reply:
I modified the comment to say "Rare, but..."